### DISCUSSION TITLE
I want to attach a function when clicking allowClear icon on Input

### DISCUSSION THREAD

**Discussion by vitalijalbu** [ant-design/ant-design#45724] [Q&A] [answered]
I want to attach a function when clicking allowClear icon on Input, like a console log.
i think it can be done with custom icon, but don't want a custom icon

**Comment by zombieJ** [comment]
Currently not provider `onClear` prop. But I think customize the icon you mentioned is a good solution.

**Reply by vitalijalbu** [reply]
i tried also with onClear but nothing... will it be released soon?

**Reply by vitalijalbu** [reply]
@zombieJ should I open a request feature for that?

**Comment by afc163** [comment, marked as answer]
https://github.com/ant-design/ant-design/issues/16862#issuecomment-507616942
